### PR TITLE
GP2-3859-Optional phone number

### PR DESCRIPTION
**Pull request by seijihg**
Updated with phone number field.
Registration:
Screenshot 2021-12-10 at 21 19 29
Bad Phone:
Screenshot 2021-12-13 at 18 04 59
Validation:
Screenshot 2021-12-13 at 18 05 15
Input good phone:
Screenshot 2021-12-13 at 18 06 05
Sucessfully:
Screenshot 2021-12-13 at 18 06 12

Profile is created with phone number:
Screenshot 2021-12-13 at 18 06 22
Screenshot 2021-12-13 at 18 06 32

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-3859
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Ran the `make manage download_geolocation_data` command
- [ ] I have updated security dependencies
- [x] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.9.5
```
Comment (bobmeredith) [inline .python-version:1]: has this changed?

File: react-components/src/components/Signup/Form.jsx [hunk 2]
```diff
@@ -90,12 +103,15 @@ Form.propTypes = {
...
     email: PropTypes.arrayOf(PropTypes.string),
     password: PropTypes.arrayOf(PropTypes.string),
+    mobile_phone_number: PropTypes.arrayOf(PropTypes.string)
   }),
   handlePasswordChange: PropTypes.func.isRequired,
   handleSubmit: PropTypes.func.isRequired,
   handleEmailChange: PropTypes.func.isRequired,
   password: PropTypes.string,
   email: PropTypes.string,
+  phoneNumber: PropTypes.string,
```
Comment (bobmeredith) [inline react-components/src/components/Signup/Form.jsx:113]: If you have a non mandatory property, it should have a default.

File: sso/helpers.py [hunk 3]
```diff
@@ -124,8 +124,10 @@ def check_verification_code(uidb64, token, code):
     return response
 
 
-def create_user(email, password):
-    response = sso_api_client.user.create_user(email=email, password=password)
+def create_user(email, password, **kwargs):
+    response = sso_api_client.user.create_user(
+        email=email, password=password, mobile_phone_number=kwargs.get('mobile_phone_number')
+    )
```
Comment (bobmeredith) [inline sso/helpers.py:130]: Small thing, but I didn't think this was specifically a mobile number. The name could confuse.

File: sso/serializers.py [hunk 4]
```diff
@@ -1,9 +1,15 @@
+from django.core.validators import RegexValidator
 from rest_framework import serializers
 
+phone_uk_regex = RegexValidator(
+    regex=r'^(?:0|\+?44)(?:\d\s?){9,10}$', message='Please enter a valid UK telephone number.'
+)
```
Comment (bobmeredith) [inline sso/serializers.py:6]: Nice regexp but looks like it won't allow a space after the '0' or '+44'. suggest move the white-spece matcher to before the digit. maybe also make it a * to match any number. `^(?:0|\+?44)(?:\s*\d){9,10}$`
Don't know if you need to match trailing spaces.

**Review by bobmeredith — COMMENTED** [review]
Looking good - but a few comments.

File: sso/helpers.py (hunk 3 above)
Comment (bobmeredith) [reply sso/helpers.py:130]: My bad - mobile_number matches the name in the model - and we're not changing that.

**Review by depsiatwal — APPROVED** [review]
LGTM

**Pull request merged**